packages/model-fragments/lib/fragments/model.js

```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
   copy: function() {
     var store = get(this, 'store');
     var type = store.modelFor(this.constructor);
-    var data = {};
-
-    // TODO: handle copying sub-fragments
-    Ember.merge(data, this._data);
-    Ember.merge(data, this._attributes);
-
-    return this.store.createFragment(type, data);
+    
+    var newFragment = store.createFragment(type);
+    get(this.constructor, 'attributes').forEach(function(attribute) {
+      if( attribute.type === 'fragment' ) {
```

Comment:
Small nit: can you make the spaces around parens consistent with the rest of the repo? e.g. `if (attribute.type === 'fragment') {`